src/hub/vendor/customer.py

```diff
@@ -143,9 +143,41 @@ def run(self) -> None:
                 last4=last4,
                 charge=charge,
             )
+            # NOTE:
+            # Firefox's route has a particular casing requirement
+            # Salesforce's route doesn't care.
+            data_projection_by_route = {
+                StaticRoutes.FIREFOX_ROUTE: [
+                    "uid",
+                    "active",
+                    "subscriptionId",
+                    "productName",
+                    "eventId",
+                    "eventCreatedAt",
+                    "messageCreatedAt",
+                ],
+                StaticRoutes.SALESFORCE_ROUTE: data.keys(),
+            }
+            subsets = []
+            for route in data_projection_by_route:
+                try:
+                    logger.debug("sending to", key=route)
+                    subset = dict(
+                        (k, data[k])
+                        for k in data_projection_by_route[route]
+                        if k in data
+                    )
+                    payload = {"type": route, "data": json.dumps(subset)}
+                    subsets.append(payload)
+                    logger.info("subset", subset=subset)
+
+                    logger.debug("sent to", key=route)
+                except Exception as e:
+                    # log something and maybe change the exception type.
+                    logger.error("projection exception", error=str(e))
```

Comment:
be consistent; all of the other `error=e` occurrences in the codebase don't require str(); are you sure this isn't the same?

Reply:
Good point, should be.

Reply:
done.